src/__integrations__/pmClient.browser.docker.test.ts

```diff
@@ -91,13 +111,29 @@ describe('Data Submission Form', () => {
     await page.waitForSelector(buttonS);
 
     const serverResponse = await Promise.all([
-      page.waitForResponse(response => response.request().method() === 'OPTIONS'),
-      page.waitForResponse(response => response.request().method() === 'POST'),
+      page.waitForResponse(
+        response => response.request().method() === 'OPTIONS' && /start-session/.test(response.request().url()),
+      ),
+      page.waitForResponse(
+        response => response.request().method() === 'OPTIONS' && /approve-terms/.test(response.request().url()),
+      ),
+      page.waitForResponse(
+        response => response.request().method() === 'OPTIONS' && /wallet/.test(response.request().url()),
+      ),
+      page.waitForResponse(
+        response => response.request().method() === 'POST' && /start-session/.test(response.request().url()),
+      ),
       page.click(buttonS),
     ]);
 
     expect(serverResponse[0]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
-    await expect(serverResponse[1]?.json()).resolves.toMatchObject({ data: { user: { email: 'pippo@pluto.com' } } });
+    expect(serverResponse[1]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
+    expect(serverResponse[2]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
+    expect(serverResponse[3]?.headers()['access-control-allow-origin']).toEqual(`http://${SRV_HOST}:${SRV_PORT}`);
+
+    await expect(serverResponse[3]?.json()).resolves.toMatchObject({
```

Comment:
Why do you check `OPTION` allow origin header for all three endpoints `start`, `approve` and `wallet` and for the last one check `payload` email ?

The purpose of test is `should call start session, approve terms and add wallet when Continua is pressed `

Reply:
@pasqualespica I agree I should have written a new test case to assert on CORS preflights. I'll refactor as you suggest